src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+
+### Part 2
+{:.no_toc}
+
+This part also contains two examples, both focusing on contact information. 
+In the first example, you'll wrap the `Row` you built in a `Column`, 
+so you can create space for the contact information. 
+In the second example, you'll enter the contact information. 
+At the end of Part 2, your code will contain a `Column` within a `Row` within a `Column`.
+
+#### Exercise: Tweak the layout
+{:.no_toc}
+{{site.alert.secondary}}
+  
+  Wrap your `Row` inside a `Column` that contains the following widgets:
+
+<ul markdown="1">
+  <li markdown="1"> 
+  A `mainAxisSize` property set to `MainAxisSize.min`. 
+  </li>
+  <li markdown="1">
+  A `crossAxisAlignment` property set to `CrossAxisAlignment.stretch`.
+  </li>
+  <li markdown="1">
+  A `SizedBox` widget with a height of 8.
+  </li>
+  <li markdown="1">
+  An empty `Row` class where you'll add your contact information. 
+  </li>
+  <li markdown="1">
+  A second `SizedBox` widget with a height of 16.
+  </li>
+  <li markdown="1">
+  A second empty `Row` class where you'll add different icons (covered in [Part 3](#part-3)). 
+  </li>
+</ul>
+
+  Your `SizedBox` widgets and empty `Row` classes should be formatted like this: 
+
+  ```dart
+
+     ],
+    ),
+    SizedBox(),
+    Row(), // First empty Row class.
+    SizedBox(),
+    Row(), // Second empty Row class.
+   ],
+  );
+
+  ```
+{{site.alert.end}}
+
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=c3ac34ed8952724a0ecb0af1445c2af8&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+#### Example: Enter contact information 
+{:.no_toc}
+{{site.alert.secondary}}
+  Enter the following widgets inside the first empty `Row` class: 
+
+<ul markdown="1">
+  <li markdown="1">  
+  A `Text` widget that contains the address "123 Main Street." 
+  </li>
+  <li markdown="1">
+  A second `Text` widget that contains the phone number "123-456-7890."
+  </li>
+  <li markdown="1"> 
+  A `mainAxisAlignment` property set to `mainAxisAlignment.SpaceBetween`.
+  </li>
+</ul>
+{{site.alert.end}}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=c5be61116652927c5d92262fce1b5360&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+### Part 3
+{:.no_toc}
+This part contains one example where you finish your business card display. You'll add four `Icon` widgets to the second empty `Row` class that you built in Part 2. 
+
+{{site.alert.secondary}}
+  Enter the following `Icon` widgets in the second empty `Row` class.
+  * `Icons.accessibility` 
+  * `Icons.timer`
+  * `Icons.phone_android`
+  * `Icons.phone_iphone`
+{{site.alert.end}}
+
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=dae36611fc9af04c4b9d0fbc3429275e&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
```

Comment:
Add some sort of summary, with info on where they might want to go next.

Reply:
The code in the Row that starts at line 29 is missing the `mainAxisAlignment: MainAxisAlignment.spaceBetween,` (line 34 in the solution) so the test fails, through no fault of the reader.